Dockerfile

```diff
@@ -42,6 +42,7 @@ WORKDIR /app
 # this resolves some chicken-and-egg problems with using workspace bins before they're created (install -> build -> install)
 RUN pnpm recursive run prepare
 RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile
+ENV NODE_OPTIONS=--max-old-space-size=4096
 RUN pnpm run -r build
```

Comment:
do you know `ENV` sticks around in the docker container? as in running the indexer would use 4gb memory too?

Reply:
https://docs.docker.com/reference/dockerfile/#env
> The ENV instruction sets the environment variable <key> to the value <value>. This value will be in the environment for all subsequent instructions in the build stage and can be [replaced inline](https://docs.docker.com/reference/dockerfile/#environment-replacement) in many as well.

sounds like just during build?

Reply:
but also seeing
> If an environment variable is only needed during build, and not in the final image, consider setting a value for a single command instead:

so I wonder if we use this instead:
```
RUN NODE_OPTIONS=--max-old-space-size=4096 pnpm run -r build
```